tests/Doctrine/Tests/ORM/Functional/ResultCacheTest.php

```diff
@@ -168,6 +167,23 @@ public function testEnableResultCache(): void
         $this->_em->getConfiguration()->setResultCacheImpl(new ArrayCache());
     }
 
+    public function testEnableResultCacheWithIterable(): void
```

Comment:
@greg0ire this is the test case.

Reply:
Wow, that's suprising, it looks like a very common scenario… kind of weird that nobody else reported the issue. Did it happen since a recent upgrade, or did you experiencing on new code you wrote?